### PR TITLE
VPN-2585 - Set up advanced tips & tricks section on tips & tricks view

### PR DESCRIPTION
**Pull request by brizental**
## Description

Added the Advanced Tips & Tricks section to the Tips & Tricks view as specific in the design: https://www.figma.com/file/LmWggJlWXCi23jOpitIDnA/VPN-Tips-%26-Tricks?node-id=1054%3A52928.

As it is right now, since there are no advanced tips or tutorials, that section will not show up at all.

**Note**: I was not able to make changes to guides and tutorials locally to test this due to [VPN-2593](https://mozilla-hub.atlassian.net/browse/VPN-2593). I plan to work on that ASAP, so I'll look for issues once that is fixed. For now, since this change actually produces no real change in the views I thought this was ok.

## Reference

https://mozilla-hub.atlassian.net/browse/VPN-2585

## Checklist
    
- [ ] My code follows the style guidelines for this project
- [x] I have not added any packages that contain high risk or unknown licenses (GPL,  LGPL, MPL, etc. consult with DevOps if in question)
- [x] I have performed a self review of my own code
- [ ] ~I have commented my code PARTICULARLY in hard to understand areas~
- [ ] ~I have added thorough tests where needed~

### CODE REVIEW

File: docs/guides.md [hunk 1]
```diff
@@ -15,36 +15,36 @@ On this page, we want to describe how to write new guides.
 The Guide JSON files are add-on JSON objects, with type "guide" and a "guide"
```
Comment (brizental) [inline docs/guides.md:15]: I am sorry for all of these Markdown formatting changes. VSCode just did that automatically on save (?). If it's a problem I can rever them.

**Comment by bakulf** [comment]
Do you want to add tests? At least unit tests to see if the advanced property is propagated correctly.

File: src/ui/settings/ViewTipsAndTricks/TipsAndTricksSection.qml [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+    property var guidesModel
+
+    property bool hasTutorials
+    property var tutorialsModel
+
+    visible: hasGuides || hasTutorials
+
+    // Title
+    VPNBoldLabel {
+        Layout.topMargin: 32
```
Comment (MattLichtenstein) [inline src/ui/settings/ViewTipsAndTricks/TipsAndTricksSection.qml:30]: nit: I think moving `Layout.topMargin: 32` to where we instantiate the TipsAndTricksSection's in ViewTipsAndTricks makes more sense from a readability perspective, and also, we don't have to dig down to the component level and make special exceptions if we ever want to change the spacing between sections

File: src/ui/settings/ViewTipsAndTricks/TipsAndTricksSection.qml [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+import QtQuick.Layouts 1.15
+import QtQuick.Controls 2.15
+
+import Mozilla.VPN 1.0
+import components 0.1
+
+ColumnLayout {
+    id: tipsAndTricksSection
+
+    property var flickableParent
```
Comment (MattLichtenstein) [inline src/ui/settings/ViewTipsAndTricks/TipsAndTricksSection.qml:15]: nit: since we're only accessing the flickable's `width` property, maybe we just pass in the width instead of the whole component

**Review by MattLichtenstein — APPROVED** [review]
This looks great! Love the componentized sections!

File: translations/strings.yaml [hunk 4]
```diff
@@ -178,7 +178,6 @@ inAppSupportWorkflow:
   supportErrorButton: Try Again
```
Comment (brizental) [inline translations/strings.yaml:178]: Again, VSCode just formatted this file on save. If that is an issue I am happy to revert the changes.

File: translations/strings.yaml [hunk 5]
```diff
@@ -596,6 +595,8 @@ tipsAndTricks:
   featureTourCardDescription: Take a tour of Mozilla VPN to learn how the latest features help you protect your online privacy.
   quickTipsTitle: Quick tips
   quickTipsDescription: Let us help you get the most out of Mozilla VPN
+  advancedTipsTitle: Advanced tips
+  advancedTipsDescription: Let us help you learn more about Mozilla VPN
```
Comment (brizental) [inline translations/strings.yaml:599]: These two lines are the actual changes that matter.

File: translations/strings.yaml [hunk 6]
```diff
@@ -683,13 +684,12 @@ global:
...
+  vpnUser:
     value: VPN User
     comment: Fallback string for users who have not specified a Display Name.
-  discoverNow: 
+  discoverNow:
     value: Discover now
     comment: Label for button used in intro modals to discover new features.
   goBack:
     value: Go back
     comment: Label for button used in intro modals to return to the app.
```
Comment (flodolo) [inline translations/strings.yaml:695]: The other changes are nice (there should be no trailing spaces in this file, and I'm all for consistency), but please keep an empty line at the end of this file (I thought we added checks for it, but I might be wrong).

**Review by flodolo — APPROVED** [review]
I don't mind

File: translations/strings.yaml (hunk 6 above)
Comment (brizental) [reply translations/strings.yaml:695]: Hum, new lines at the end of files is actually I something my VSCode _should_ be setup to add on save 🤔  Anyways, I'll re-add it.

Comment (flodolo) [reply translations/strings.yaml:695]: Might something related to a package for YAML? I don't think VS Code reformats YAML out of the box.

File: src/ui/settings/ViewTipsAndTricks/ViewTipsAndTricks.qml [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+            VPNFilterProxyModel {
+                id: quickTutorialsModel
+                source: VPNAddonManager
+                filterCallback: ({ addon }) => isQuickAddon(addon, "tutorial")
+            }
+
+            TipsAndTricksSection {
+                Layout.topMargin: 32
+
+                id: quicksTips
```
Comment (MattLichtenstein) [inline src/ui/settings/ViewTipsAndTricks/ViewTipsAndTricks.qml:112]: the smallest of small nits: the [qml coding convetions](https://doc.qt.io/qt-6/qml-codingconventions.html) say to put the id property at the top of the object block

It is totally fine the way it is, I'm just trying to give you a warm welcome to qml 😄

File: translations/strings.yaml (hunk 6 above)
Comment (brizental) [reply translations/strings.yaml:695]: Hum, it's the prettier formatter. It now supports not only JavaScript but also Markdown and YAML apparently. That explains all these reformattings.

Anyways, so this is fun. Maybe we found a bug in git? So here is what is happening: there _is_ actually a new line in my local file, VSCode always adds a new line when I save and trims extra new lines. I think git is confused, because there were trailing spaces there which were removed. See:

Screenshot 2022-07-21 at 17 15 05

I sent a commit adding the newline: https://github.com/mozilla-mobile/mozilla-vpn-client/pull/4046/commits/e28d0fc58ebf1a6b420feac1b0d2e82e75d19159, and now git is happy.

However in order to do that I had to disable my formatter on VSCode, because that will strip extra new lines... And probably will have to remember to do that always when editing this file, because even though it now looks corerct for git, this is how it looks for me:

Screenshot 2022-07-21 at 17 34 21

YAY!
